NetKet/Hamiltonian/MatrixWrapper/direct_matrix_wrapper.hpp

```diff
@@ -26,24 +31,25 @@ namespace netket {
  * computed from Operator::FindConn every time Apply is called.
  */
 template <class Operator, class WfType = Eigen::VectorXcd>
-class DirectMatrixWrapper : public AbstractMatrixWrapper<WfType> {
+class DirectMatrixWrapper : public AbstractMatrixWrapper<Operator, WfType> {
   const Operator& operator_;
+  HilbertIndex hilbert_index_;
```

Comment:
Then maybe we should store also Hilbert & hilbert_; here @femtobit

Reply:
Since this is just a reference, storing it has no noticeable performance impact so I think this is not necessary.

Reply:
True, agreed